model/src/main/java/org/cloudfoundry/identity/uaa/zone/Links.java

```diff
@@ -108,6 +111,30 @@ public boolean isSelfServiceLinksEnabled() {
 
         public SelfService setSelfServiceLinksEnabled(boolean selfServiceLinksEnabled) {
             this.selfServiceLinksEnabled = selfServiceLinksEnabled;
+            if(!selfServiceLinksEnabled) {
+                this.selfServiceCreateAccountEnabled = false;
+                this.selfServiceResetPasswordEnabled = false;
+            }
+            return this;
+        }
+        public SelfService setSelfServiceCreateAccountEnabled(boolean selfServiceCreateAccountEnabled) {
+            if(selfServiceCreateAccountEnabled){
+                this.signup = "/create_account";
```

Comment:
Hence setSignUp look for a value and response return selfServiceCreateAccountEnabled= false for a case, for which two of the cases are falling,  because of which I have explicitly set the value during selfServiceCreateAccountEnabled.

Reply:
What tests are failing if you remove these lines?

Reply:
Below are the tests failing if those lines are removed.

1. uaac -t curl -X POST /identity-zones -H 'Content-Type: application/json' -d"{ \"id\": \"uaa-zone-6\", \
          \"subdomain\":\"uaa-zone-6\", \"name\":\"uaa-zone-6\",
           \"config\" : { \"links\" : { \"selfService\" : {
           \"selfServiceResetPasswordEnabled\" : true,
           \"selfServiceCreateAccountEnabled\" : true } } } 
        }"

2. uaac -t curl -X POST /identity-zones -H 'Content-Type: application/json' -d"{ \"id\": \"uaa-zone-8\", \
          \"subdomain\":\"uaa-zone-8\", \"name\":\"uaa-zone-8\",
           \"config\" : { \"links\" : { \"selfService\" : {
           \"selfServiceResetPasswordEnabled\" : false,
           \"selfServiceCreateAccountEnabled\" : true } } }
        }"

Reply:
That's a design decision then as you are changing behavior compare to the legacy self-service flag and links.

Currently, when the legacy self-service flag is enabled and signup link is not provided, "create account" link is not displayed.  However, user can navigate to /create_account page and do POST to create account endpoint.

We have two choices:
1. If selfServiceCreateAccountEnabled = true and signup link is not specified, do not display "create account" link.  User should NOT be able to navigate to /create_account page and do POST to create account endpoint.
2. If selfServiceCreateAccountEnabled = true and signup link is not specified, use default /create_account link the same way it's done for /forgot_password.  Then do it not at the model level, but in [LoginInfoEndpoint.java#L980-L981](https://github.com/GESoftware-CF/uaa/blob/dd3673f0f68439c40eb76feb0b4ab6348aacf0d0/server/src/main/java/org/cloudfoundry/identity/uaa/login/LoginInfoEndpoint.java#L980-L981)

Reply:
Option 2 might be more straight forward and preferable...

Reply:
As discussed, I have updated If selfServiceCreateAccountEnabled = true and signup link is not specified, will have default /create_account link else the custom one.